message_handlers/tx_handler.py

```diff
@@ -35,6 +37,70 @@ def onAckNak(packet):
 
     draw_messages_window()
 
+def on_response_traceroute(packet):
+    """on response for trace route"""
+    from ui.curses_ui import draw_channel_list, draw_messages_window, add_notification
+
+    UNK_SNR = -128 # Value representing unknown SNR
+
+    route_discovery = mesh_pb2.RouteDiscovery()
+    route_discovery.ParseFromString(packet["decoded"]["payload"])
+    msg_dict = google.protobuf.json_format.MessageToDict(route_discovery)
+
+    msg_str = "Route traced towards destination:\n"
+
+    route_str = get_name_from_number(packet["to"]) or f"{packet['to']:08x}" # Start with destination of response
+
+    # SNR list should have one more entry than the route, as the final destination adds its SNR also
+    lenTowards = 0 if "route" not in msg_dict else len(msg_dict["route"])
+    snrTowardsValid = "snrTowards" in msg_dict and len(msg_dict["snrTowards"]) == lenTowards + 1
+    if lenTowards > 0: # Loop through hops in route and add SNR if available
+        for idx, node_num in enumerate(msg_dict["route"]):
+            route_str += " --> " + (get_name_from_number(node_num) or f"{node_num:08x}") \
+                     + " (" + (str(msg_dict["snrTowards"][idx] / 4) if snrTowardsValid and msg_dict["snrTowards"][idx] != UNK_SNR else "?") + "dB)"
+
+    # End with origin of response
+    route_str += " --> " + (get_name_from_number(packet["from"]) or f"{packet['from']:08x}") \
+             + " (" + (str(msg_dict["snrTowards"][-1] / 4) if snrTowardsValid and msg_dict["snrTowards"][-1] != UNK_SNR else "?") + "dB)"
+
+    msg_str += route_str + "\n" # Print the route towards destination
+
+    # Only if hopStart is set and there is an SNR entry (for the origin) it's valid, even though route might be empty (direct connection)
+    lenBack = 0 if "routeBack" not in msg_dict else len(msg_dict["routeBack"])
+    backValid = "hopStart" in packet and "snrBack" in msg_dict and len(msg_dict["snrBack"]) == lenBack + 1
+    if backValid:
+        msg_str += "Route traced back to us:\n"
+        route_str = get_name_from_number(packet["from"]) or f"{packet['from']:08x}" # Start with origin of response
+
+        if lenBack > 0: # Loop through hops in routeBack and add SNR if available
+            for idx, node_num in enumerate(msg_dict["routeBack"]):
+                route_str += " --> " + (get_name_from_number(node_num) or f"{node_num:08x}") \
+                         + " (" + (str(msg_dict["snrBack"][idx] / 4) if msg_dict["snrBack"][idx] != UNK_SNR else "?") + "dB)"
+
+        # End with destination of response (us)
+        route_str += " --> " + (get_name_from_number(packet["to"]) or f"{p['to']:08x}") \
```

Comment:
```suggestion
        route_str += " --> " + (get_name_from_number(packet["to"]) or f"{packet['to']:08x}") \
```
I was getting an error here, is this what you meant?

Reply:
Ah yup that's what it should have been!